### PR TITLE
[CPU] Xor: convert to LogicalXor

### PR DESCRIPTION
**Pull request by rbaranchuk-capgemini**
### Details:
 - conversion from Xor to LogicalXor

### Ticket: 
- CVS-93338

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@rbaranchuk-capgemini I would propose to enable this transformation by default so all plugins can benefit from it without necessity to change plugin's code.

**Comment by dmitry-gorokhov** [comment]
@itikhono could you please review the PR?

**Comment by dmitry-gorokhov** [comment]
@vladimir-paramuzov @nkogteva FYI

**Comment by rbaranchuk-capgemini** [comment]
> @rbaranchuk-capgemini I would propose to enable this transformation by default so all plugins can benefit from it without necessity to change plugin's code.

moved the conversion to common_optimizations

File: src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#include "transformations/op_conversions/convert_xor_to_logical_xor.hpp"
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/opsets/opset1.hpp>
+
+#include "itt.hpp"
+
+ov::pass::ConvertXorToLogicalXor::ConvertXorToLogicalXor() {
```
Comment (itikhono) [inline src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp:13]: could you add a test?

File: src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    auto xor_v1 = pattern::wrap_type<ov::opset1::Xor>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto xor_v1_node = std::dynamic_pointer_cast<ov::opset1::Xor>(m.get_match_root());
+        if (!xor_v1_node)
+            return false;
+
+        const auto& autobroad = xor_v1_node->get_autob();
+
+        auto logical_xor_v1 = std::make_shared<ov::opset1::LogicalXor>(xor_v1_node->input_value(0),
```
Comment (itikhono) [inline src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp:25]: nitpick: it's better to use the latest opset (opset10) version for LogicalXor

**Review by itikhono — CHANGES_REQUESTED** [review]

File: src/common/transformations/include/transformations/op_conversions/convert_xor_to_logical_xor.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ */
+class ov::pass::ConvertXorToLogicalXor : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("ConvertXorToLogicalXor", "0");
+    ConvertXorToLogicalXor();
+};
+
+namespace ngraph {
+namespace pass {
+using ov::pass::ConvertXorToLogicalXor;
```
[the remark this answers is not in the document]
Comment (rbaranchuk-capgemini) [reply src/common/transformations/include/transformations/op_conversions/convert_xor_to_logical_xor.hpp]: done

File: src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp (hunk 1 above)
Comment (rbaranchuk-capgemini) [reply src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp:14]: test is added

File: src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp (hunk 2 above)
Comment (rbaranchuk-capgemini) [reply src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp]: done

File: src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#include <ngraph/opsets/opset10.hpp>
+#include <ngraph/pass/manager.hpp>
+#include <string>
+#include <transformations/init_node_info.hpp>
+#include <transformations/op_conversions/convert_xor_to_logical_xor.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+using namespace testing;
+using namespace ngraph;
```
Comment (itikhono) [inline src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp:19]: please use "ov" namespace

File: src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset1.hpp>
```
Comment (itikhono) [inline src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp:9]: please use openvino headers
<openvino/...

File: src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        if (!xor_v1_node)
+            return false;
+
+        const auto& autobroad = xor_v1_node->get_autob();
+
+        auto logical_xor_v10 = std::make_shared<ov::opset10::LogicalXor>(xor_v1_node->input_value(0),
+                                                                         xor_v1_node->input_value(1),
+                                                                         autobroad);
+        logical_xor_v10->set_friendly_name(xor_v1_node->get_friendly_name());
+        ngraph::copy_runtime_info(xor_v1_node, logical_xor_v10);
```
Comment (itikhono) [inline src/common/transformations/src/transformations/op_conversions/convert_xor_to_logical_xor.cpp:30]: please use "ov" namespace

**Review by itikhono — APPROVED** [review]
looks good but please fix minor comments

File: src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp (hunk 5 above)
Comment (rbaranchuk-capgemini) [reply src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp]: done

File: src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp (hunk 4 above)
Comment (rbaranchuk-capgemini) [reply src/tests/functional/inference_engine/transformations/op_conversions/convert_xor_to_logical_xor.cpp]: done

**Review by ceciliapeng2011 — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**